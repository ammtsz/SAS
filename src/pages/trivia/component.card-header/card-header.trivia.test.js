import React from "react";
import CardHeaderTrivia from "./card-header.trivia.component";

import { render, screen, cleanup } from "@testing-library/react";
import configureMockStore from "redux-mock-store";
import { Provider } from "react-redux";

describe("component trivia-page <CardHeaderTrivia />", () => {
  const mockStore = configureMockStore();
  const mockQuestionNumber = 3;
  const store = mockStore({
    quiz: {
      difficulty: "medium",
      questionNumber: mockQuestionNumber,
    },
  });

  let component;

  beforeEach(() => {
    component = render(
      <Provider store={store}>
        <CardHeaderTrivia />
      </Provider>
    );
  });

  afterEach(() => {
    cleanup();
  });

  it("should render CardHeaderTrivia", () => {
    expect(component.asFragment()).toMatchSnapshot();
  });

  it("should print question number", () => {
    expect(
      screen.getByText(`Question ${mockQuestionNumber}`)
    ).toBeInTheDocument();
  });

  it("should print 2 stars for 'medium' difficulty", () => {
    expect(screen.getByAltText("star-level-easy")).not.toHaveClass("starless");
    expect(screen.getByAltText("star-level-medium")).not.toHaveClass("starless");
    expect(screen.getByAltText("star-level-hard")).toHaveClass("starless");
  });

  it("should contain 'Medium' text", () => {
    expect(screen.queryByText("Easy")).toEqual(null);
    expect(screen.queryByText("Medium")).toBeInTheDocument();
    expect(screen.queryByText("Hard")).toEqual(null);
  });
});
