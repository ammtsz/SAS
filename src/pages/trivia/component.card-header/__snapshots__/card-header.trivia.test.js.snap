// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component trivia-page <CardHeaderTrivia /> should render CardHeaderTrivia 1`] = `
<DocumentFragment>
  <div
    class="sc-bdfBwQ gpnPqW"
  >
    <h6
      class="sc-gsTCUz jiOGPV"
      tabindex="0"
    >
      Question 3
    </h6>
    <div
      class="sc-dlfnbm jrYdKu"
    >
      <span
        class="sc-hKgILt dFTvLt"
        tabindex="-1"
      >
        <img
          alt="star-level-easy"
          class=""
          src="star.svg"
        />
        <img
          alt="star-level-medium"
          class=""
          src="star.svg"
        />
        <img
          alt="star-level-hard"
          class="starless"
          src="star.svg"
        />
      </span>
      <span
        tabindex="0"
      >
        Medium
      </span>
    </div>
  </div>
</DocumentFragment>
`;
